backend/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/ibeckermayer/teleport-interview/backend/internal/handlers"
+)
+
+func main() {
+	router := mux.NewRouter()
+
+	router.Handle("/api/login", http.HandlerFunc(handlers.Login)).Methods("POST")
+
+	// NOTE: It's important that this handler be registered after the other handlers, or else
+	// all routes return a 404 (at least in development). TODO: figure out why this is the case.
+	router.PathPrefix("/").Handler(handlers.NewSpaHandler("../frontend", "index.html"))
+
+	log.Println("Server listening on port 8000")
+	log.Fatal(http.ListenAndServeTLS(":8000", "../certs/localhost.crt", "../certs/localhost.key", router))
+}
```

Comment:
I wouldn't include this logic in `main`. Instead create a package that has your server in it and start that instead. Ideally your `main` would just be flag parsing and then calling that package to start the server. Maybe doing some signal handling but I think that's out of scope for your task.

Reply:
Gotcha, will refactor this into something cleaner/extensible.